File: go/2405_Optimal_Partition_of_String/2405_Optimal_Partition_of_String.go
func partitionString(s string) int {
    count := 1
    var m [26]int
    for _, c := range s {
        if m[c - 'a'] == 1 {
            count += 1
            m = [26]int{}
        }
        m[c - 'a'] += 1
    }
    return count
}

File: java/2405_Optimal_Partition_of_String/2405_Optimal_Partition_of_String.java
class Solution {
    public int partitionString(String s) {
        int count = 1;
        int[] set = new int[26];
        for (char c : s.toCharArray()) {
            if (set[c-'a'] == 1) {
                count += 1;
                set = new int[26];
            }
            set[c-'a'] += 1;
        }
        return count;
    }
}
